op-node/rollup/derive/calldata_source.go

```diff
@@ -104,21 +120,28 @@ func (ds *DataSource) Next(ctx context.Context) (eth.Data, error) {
 // DataFromEVMTransactions filters all of the transactions and returns the calldata from transactions
 // that are sent to the batch inbox address from the batch sender address.
 // This will return an empty array if no valid transactions are found.
-func DataFromEVMTransactions(config *rollup.Config, batcherAddr common.Address, txs types.Transactions, log log.Logger) []eth.Data {
+func DataFromEVMTransactions(config *rollup.Config, batcherAddr common.Address, txs types.Transactions, receipts types.Receipts, log log.Logger) []eth.Data {
 	var out []eth.Data
 	l1Signer := config.L1Signer()
 	for j, tx := range txs {
 		if to := tx.To(); to != nil && *to == config.BatchInboxAddress {
+			receipt := receipts[j]
```

Comment:
Why do we need "seqDataSubmitter, err := l1Signer.Sender(tx) // optimization: only derive sender if To is correct"?
Should not the transaction get reverted if the tx is not signed by the right batcher?

Reply:
Good question. We may also be able to skip this if the version is 1. I think it should definitely be kept for version 0 though.